src/js/common/components/RateLimiter.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import LoadingIndicator from './LoadingIndicator';
+
+/*
+ * Expects a settings object that looks like:
+ *
+ * settings: {
+ *   [id]: {
+ *     rateLimitAuthed: 1,
+ *     rateLimitUnauthed: 1
+ *   }
+ * }
+ */
+export class RateLimiter extends React.Component {
+  constructor(props) {
+    super(props);
+    // 0 is nothing gets through, 1 is everything gets through
+    const { rateLimitAuthed = 1, rateLimitUnauthed = 1 } = window.settings[props.id] || {};
```

Comment:
Could we make this more consistent by either replacing `{}` with `{ rateLimitAuthed: 1, rateLimitUnauthed: 1 }` or removing the default values?

Reply:
That's not quite the same thing. It would end up with undefined values when there's a vic object in the settings, but no rate limit properties. I don't want to assume that nothing else will go in settings objects created for particular apps.